exchanges/subscription/list.go

```diff
@@ -91,7 +92,12 @@ func fillAssetPairs(ap assetPairs, a asset.Item, e iExchange) error {
 
 // assetPairs returns a map of enabled pairs for the subscriptions in the list, formatted for the asset
 func (l List) assetPairs(e iExchange) (assetPairs, error) {
-	at := e.GetAssetTypes(true)
+	at := []asset.Item{}
+	for _, a := range e.GetAssetTypes(true) {
+		if e.IsAssetWebsocketSupported(a) {
```

Comment:
With only `Index` asset enabled
Only trade sub enabled:
```
 {
      "enabled": true,
      "channel": "trade",
      "asset": "all"
     },
```

```
[ERROR] | WEBSOCKET | 14/11/2024 11:49:43 | Bitmex websocket: subscription failure, trade all : subscription template did not generate the expected number of asset records: Got 1; Expected 0
```

Do you classify this as something to address?

Reply:
Absolutely. That's a bug in the sub template (which I've made a few times now).
I'll fix.

Reply:
Note:
`Bitmex asset type index websocket functionality is unsupported, REST fetching only.`